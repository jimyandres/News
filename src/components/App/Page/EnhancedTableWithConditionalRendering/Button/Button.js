import React from 'react';
import PropTypes from 'prop-types';
import './index.css';
import MaterialButton from 'material-ui/Button';
import { withStyles } from 'material-ui/styles';

const style = {
  label: {
    textDecoration: 'underline overline',
  },
};

const Button = ({onClick, className, children, active = false, classes, ...rest}) =>
  <MaterialButton
    onClick={onClick}
    color='primary'
    classes={{ label: active ? classes.label : ''}}
    {...rest}
  >
    {children}
  </MaterialButton>

Button.defaultProps = {
  className: '',
}

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};

export default withStyles(style)(Button);
