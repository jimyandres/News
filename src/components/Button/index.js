import React from 'react';
import PropTypes from 'prop-types';
import FlatButton from 'material-ui/FlatButton';
import './index.css';

const Button = ({onClick, className, children}) =>
  <FlatButton
    onClick={onClick}
    className={className}
    type="button"
    primary={true}
  >
    {children}
  </FlatButton>

Button.defaultProps = {
  className: '',
}

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};

export default Button;
