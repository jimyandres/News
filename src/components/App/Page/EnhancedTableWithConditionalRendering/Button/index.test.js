import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {

  const somefunc = () => "something";

  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button onClick={somefunc}>Testing</Button>, div);
  });

  test('has a valid snapshot', () => {
    const component =  renderer.create(
      <Button onClick={somefunc}>Testing</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render children when passed in', () => {
    const element = shallow(
      <Button onClick={somefunc}>
        Search
      </Button>
    );

    expect(element.contains('Search')).toEqual(true);
  });
});
