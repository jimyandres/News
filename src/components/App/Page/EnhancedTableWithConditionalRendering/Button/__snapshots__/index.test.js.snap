// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `
<button
  className="MuiButtonBase-root-17 MuiButton-root-2 MuiButton-flatPrimary-4"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    Testing
  </span>
  <span
    className="MuiTouchRipple-root-20"
  />
</button>
`;
